docs/reference/search/suggesters/context-suggest.asciidoc

```diff
@@ -1,237 +1,269 @@
+completion field.
+Every context mapping has a unique name and a type. Currently, there are two types
+*category* and *geo*. The context mappings is configured under the `contexts` parameter
+in the type mapping.
+
+The following defines two context mappings for a completion field:
 
 [source,js]
 --------------------------------------------------
-PUT services/_mapping/service
+PUT place/shops/_mapping
 {
-    "service": {
-        "properties": {
-            "name": {
-                "type" : "string"
-            },
-            "tag": {
-                "type" : "string"
-            },
-            "suggest_field": {
-                "type": "completion",
-                "context": {
-                    "color": { <1>
+  "shops" : {
+        "properties" : {
+            ...
+            "suggest" : {
+                "type" : "completion",
+                "contexts": [
+                    { <1>
+                        "name": "place_type",
                         "type": "category",
-                        "path": "color_field",
-                        "default": ["red", "green", "blue"]
+                        "path": "cat"
                     },
-                    "location": { <2>
+                    { <2>
+                        "name": "location"
                         "type": "geo",
-                        "precision": "5m",
-                        "neighbors": true,
-                        "default": "u33"
+                        "path": "loc"
                     }
-                }
+                ]
             }
         }
     }
 }
 --------------------------------------------------
-<1> See <<suggester-context-category>>
-<2> See <<suggester-context-geo>>
+<1> Defines a `category` context named 'place_type', which will index values from field 'cat'.
+    See <<suggester-context-category>>
+<2> Defines a `geo` context named 'location', which will index values from field 'loc'.
+    See <<suggester-context-geo>>
 
-However contexts are specified (as type `category` or `geo`, which are discussed below), each
-context value generates a new sub-set of documents which can be queried by the completion
-suggester. All three types accept a `default` parameter which provides a default value to use
-if the corresponding context value is absent.
-
-The basic structure of this element is that each field forms a new context and the fieldname
-is used to reference this context information later on during indexing or querying. All context
-mappings have the `default` and the `type` option in common. The value of the `default` field
-is used, when ever no specific is provided for the certain context. Note that a context is
-defined by at least one value. The `type` option defines the kind of information hold by this
-context. These type will be explained further in the following sections.
+NOTE: Adding context mappings increases the index size for completion field.
```

Comment:
I'd at least add a link to the completion stats, which people can use to check how much memory is being used.